### ISSUE TITLE
What this socket session warning message is about for brand new Sails app (logging - verbose)

### ISSUE THREAD

**Issue by alexeygt** [balderdashy/sails#3075]
Hello, Sails!

I noticed this strange message when running **sails lift** for newly created sails app. It is shown only if logging is set to **verbose**.

**verbose:** *Could not fetch session, since connecting socket has no cookie (is this a cross-origin socket?) Generated a one-time-use cookie:sails.sid=s%3ATMAHZkeJJglAJMrautjpniCktlLldfka. %2BnnqW2a0f6RTAsCevZCGtUVzYuRIXe19FEZqMnAvfeUand saved it on the socket handshake.
This will start this socket off with an empty session, i.e. (req.session === {}) That "anonymous" section will only last until the socket is disconnected unless you persist the session id in your database,
or by setting the set-cookie response header for an HTTP request that you know came from the same user (etc) Alternatively, just make sure the socket sends a cookie header or query param when it initially connects.*

I'm very confused since this message is shown right on sails lift. There are no clients connected to the app and there should be no socket connecting.

Can somebody shed some light on this topic? I'm quite new to socket connection authorization and socket sessions. Ideal solution would be - understand why I'm getting this message and what shall I add to my application to handle this situation correctly. I don't want just to get rid of the message because I will need to authorize all my socket connections in the future. I need to understand the logic behind.

Any hints are welcome.

**Comment by PedroEscudero** [comment]
I fix it disabling sockets (in .sailsrc):

"hooks": {
    "sockets": false,
    "pubsub": false 
  }

**Comment by alexeygt** [comment]
Thank you for the reply!

The problem is - we use sockets in our WebApp, thus we cannot just disable
them.

Regards,
Alexey

**Comment by sailsbot** [comment]
Thanks for posting, @alexeygt.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
